src/dev/typescript/project.ts

```diff
@@ -26,6 +26,10 @@ function testMatchers(matchers: IMinimatch[], path: string) {
   return matchers.some((matcher) => matcher.match(path));
 }
 
+const parentProjectFactory = memoize(function (parentConfigPath: string) {
```

Comment:
all the plugin config extend `tsconfig.project.json`, no need to recreate it 123 times.